File: src/api/MaterialAdvisor/MaterialAdvisor.Application/Models/Topics/TopicListItem.cs
namespace MaterialAdvisor.Application.Models.Topics;

public class TopicListItem<TKnowledgeChecks>
{
    public Guid Id { get; set; }

    public uint Version { get; set; }

    public IEnumerable<LanguageText> Name { get; set; } = [];

    public IEnumerable<TKnowledgeChecks> KnowledgeChecks { get; set; } = [];
}

File: src/app/material-advisor/src/app/models/topic/TopicListItem.ts
import { LanguageText } from "@shared/models/LanguageText";
import { GUID } from "@shared/types/GUID";

export interface TopicListItem<TKnowledgeCheck> {
  id: GUID;
  version: number;
  owner: string;
  name: LanguageText[];
  knowledgeChecks: TKnowledgeCheck[];
  generatedAt?: Date;
}